docs/products/kafka/concepts/log-compaction.rst

```diff
@@ -1,7 +1,194 @@
+
+For example, if there is a topic containing a user's home address, on every update, a message is sent using ``user_id`` as the primary key and home address as the value:
+
+::
+
+   1001 -> "4 Privet Dr"
+   1002 -> "221B Baker Street"
+   1003 -> "Milkman Road"
+   1002 -> "21 Jump St"
+   1001 -> "Paper St"
+   1001 -> "Paper Road 21"
+
+
+There are three different options to define for how long to retain the messages:
+
+* **infinite message retention**: all changes to user's address are maintained in the logs. This can lead to the log growing in size without a bound. This option involves the risk of outgrowing the disk capacity.
+* **simple message retention**: older records are deleted after they reach a certain age or size.
+* **compacted topic**: only latest version of the key's value is kept. In the example above, only the current address for a specific user is kept.
+
+With compacted topics, Apache Kafka removes any records from the topics for which there is a newer version (based on the record key) is available in the partition. This retention policy can be set per-topic, so a single cluster can have some topics where retention is enforced by size or time and other topics where retention is enforced by compaction.
+
+Compacted topic example
+-----------------------
+
+To understand better how compaction works, we will look at a partition of a compacted topic before and after compaction has been applied.
+
+Continuing the example above, the topic records before the compaction would be:
+
+.. list-table::
+  :header-rows: 1
+  :stub-columns: 1
+  :align: left
+
+  * - Offset
+    - Key
+    - Value
+  * - 1
+    - 1001 
+    - 4 Privet Dr
+  * - 2
+    - 1002
+    - 221B Baker Street
+  * - 3
+    - 1003
+    - Milkman Road
+  * - 4
+    - 1002
+    - 21 Jump St
+  * - 5
+    - 1001
+    - Paper St
+  * - 6
+    - 1001
+    - Paper Road 21
+ 
+You can notice that there are some records with duplicate keys (``1001`` and ``1002``), with the records having offset ``4``, ``5``, and ``6`` being the addresses updates. When applying compaction, we only keep records with the latest offset (newest values) and the older ones get discarded. The end result is the following:
+
+.. list-table::
+  :header-rows: 1
+  :stub-columns: 1
+  :align: left
+
+  * - Offset
+    - Key
+    - Value
+  * - 3
+    - 1003
+    - Milkman Road
+  * - 4
+    - 1002
+    - 21 Jump St
+  * - 6
+    - 1001
+    - Paper Road 21
+
+Compacted topic details
+-----------------------
+
+A compacted topic consists of an head and a tail:
+
+* the **head** is a traditional Apache Kafka topic where new records are appended. Therefore, the head can contain duplicated keys.
+* the **tail** contains one record per key. Apache Kafka compaction ensures that keys are unique in the tail. 
+
+.. Warning:: 
```

Comment:
This seems really useful, can we put it a bit further up the post? Getting the keys right does seem fairly fundamental!

Reply:
I've moved the warning to the end of the article, this is what you meant?